src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Collections/SegmentedArrayHelper.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.Shared.Collections
+{
+    internal static class SegmentedArrayHelper
+    {
+        internal static int CalculateSegmentSize(int elementSize)
+        {
+            const int Threshold = 85000;
```

Comment:
@jkotas is it possible to detect this threshold in code?

Reply:
for testing purposes, it might be nice to eb able to confiure this so you can really stress test your system and ensure it works properly on all the boundary points.

Reply:
@CyrusNajmabadi The threshold is configurable. I'm wondering if the effective threshold can be detected in code.

Reply:
It is not possible.